assembl/static/js/app/views/tokenVoteSession.js

```diff
@@ -1708,19 +1719,107 @@ var TokenVoteSessionModal = Backbone.Modal.extend({
   }
 });
 
-var TokenVoteSessionSubmittedModal = Backbone.Modal.extend({
-  constructor: function TokenVoteSessionSubmittedModal(){
+
+// This view shows the whole vote popin and its contents
+// The model which should be given as parameter of this view is a Widget.Model instance (or a subclass of it)
+var TokenVoteSessionModal = Backbone.Modal.extend({
+  constructor: function TokenVoteSessionModal() {
     Backbone.Modal.apply(this, arguments);
   },
 
   template: '#tmpl-modalWithoutIframe',
-  className: 'modal-token-vote-session-submitted popin-wrapper',
+  className: 'modal-token-vote-session popin-wrapper',
   cancelEl: '.close, .js_close',
 
+  initialize: function(options) {
+    var that = this;
+
+    that.throttledScroll = _.throttle(that.myOnScroll, 100);
+  },
+
+  beforeCancel: function(){
+    window.location = Ctx.getDiscussionGenericURL();
+  },
+
   onShow: function(){
     var that = this;
-    var container = this.$el.find(".js_modal-body");
-    container.empty();
+
+    var displayRegion = new Marionette.Region({
+      el: that.$(".popin-body")
+    });
+
+    var outerHeader = this.$(".popin-header");
+    var displayView = new TokenVoteSessionView(
+      {
+        model: this.model,
+        showConfirmationInModal: true,
+        outerHeader: outerHeader
+      }
+    );
+
+    displayRegion.show(displayView);
+
+    that.availableTokensPositionTop = that.$(".available-tokens").position().top;
+    that.$(".available-tokens").width(that.$(".popin-body").width());
+    var popinHeaderTotalHeight = that.$(".popin-header").outerHeight();
+    that.$(".available-tokens").css("top", popinHeaderTotalHeight + (popinHeaderTotalHeight > 0 ? "px" : ""));
+    that.$(".available-tokens-container").css('min-height', "36px");
+
+    that.$(".popin-body").on('scroll', _.bind(that.onScroll, that)); // scroll event does not bubble up, and scrollable element is now .popin-body instead of this.$el
+  },
+
+  onScroll: function(){
+    this.throttledScroll();
+  },
+
+  myOnScroll: function(){
+    if (this.$(".popin-body").scrollTop() > this.availableTokensPositionTop) {
+      this.$(".available-tokens").addClass("fixed");
+    }
+    else {
+      this.$(".available-tokens").removeClass("fixed");
+    }
+  },
+
+  serializeData: function() {
+    return {
+      modal_title: i18n.gettext("Collective votes"),
+    };
+  },
+
+  onDestroy: function(){
+    if ( this.model ){
+      /*
+      We re-fetch widget data from the server, so that when the user opens the vote popin again,
+      up-to-date information is displayed (because this widget data comes from a model contained
+      in the collection collectionManager::getAllWidgetsPromise()).
+      TODO: This is a bit hacky, maybe we should find a better way, for example when popin opens, re-load user votes (instead of widget) and use these instead of the ones listed in the widget.
+      */
+      this.model.fetch();
```

Comment:
Yeah, I don't think this is the best place to do it. You should re-fetch the model from the `collectionManager` at a view initialization. If all votes have to be updated again, you can force a re-fetch on every widget modal init view. This is sprinkling more code, but more robust that this, as view removal is locked.

Reply:
This code was there before and refactoring it was not part of this task (only visual improvements), and applies only to the modal view version of it, not the full page view that is the one that will really be used.